cli/lsp/documents.rs

```diff
@@ -229,6 +193,27 @@ impl DocumentCache {
       .collect()
   }
 
+  pub fn set_dependencies(
+    &mut self,
+    specifier: &ModuleSpecifier,
+    maybe_dependencies: Option<HashMap<String, analysis::Dependency>>,
+  ) -> Result<(), AnyError> {
+    if !self.contains(specifier) {
```

Comment:
It'd be slightly more neat and efficient to handle this as the `None` case from `get_mut()`.